backend/app/repository/github_provider.py

```
@@ -675,4 +675,131 @@ async def _get_all_repositories_from_cache(
+            headers = {
+                "Authorization": f"token {git_token}",
+                "Accept": "application/vnd.github.v3+json"
+            }
+
+            # Get compare API response
+            # GitHub API format: base...head
+            # base = target_branch (what we're comparing against, e.g., main)
+            # head = source_branch (what has the changes, e.g., feature branch)
+            # This shows what changes are in source_branch that are not in target_branch
+            self.logger.info(f"Comparing {repo_name}: {target_branch}...{source_branch}")
+            response = requests.get(
```

Ruff (0.14.4)

Probable use of `requests` call without timeout

(S113)